pkg/config/apm.go

```diff
@@ -55,6 +55,8 @@ func setupAPM(config Config) {
 	config.BindEnvAndSetDefault("apm_config.windows_pipe_security_descriptor", "D:AI(A;;GA;;;WD)", "DD_APM_WINDOWS_PIPE_SECURITY_DESCRIPTOR") //nolint:errcheck
 
 	config.BindEnv("apm_config.receiver_timeout", "DD_APM_RECEIVER_TIMEOUT")                             //nolint:errcheck
```

Comment:
The `apm_config.receiver_timeout` doesn't seem to be used anymore anywhere in the codebase, should we remove it here?
This raises another question: if already used by someone, isn't that an issue to replace it this way? Do you think the CHANGELOG will be enough? If so, we should probably mention how to migrate from one to the other in the CHANGELOG for instance.

Reply:
It's still used here: https://github.com/DataDog/datadog-agent/blob/cimi/PROF-2503-increase-apm-http-server-timeout/pkg/trace/config/apply.go#L350-L354

My intention is to have the two more specific configurations (read and write) override the generic one when set.

Reply:
It is used: https://github.com/DataDog/datadog-agent/pull/6976/files#diff-cf6f1068cf79cfbcdb88f347f0b847a7ea377be2bb432081c6ebdbc7250d9166R351